File: api/health.js
// Health check endpoint for PRP Calculator API

export default function handler(req, res) {
    // Set CORS headers
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
    res.setHeader('Content-Type', 'application/json');
    
    // Handle CORS preflight
    if (req.method === 'OPTIONS') {
        res.status(200).json({ status: 'ok' });
        return;
    }
    
    // Return health status
    res.status(200).json({
        status: 'healthy',
        service: 'PRP Calculator API',
        version: '1.0.0',
        timestamp: new Date().toISOString()
    });
}

File: api/health.py
#!/usr/bin/env python3
"""
PRP Dosage Calculator API - Health Check Endpoint
Vercel serverless function for health monitoring
"""

from flask import jsonify


def handler(request):
    """
    Vercel serverless function handler for the health check endpoint
    """
    response = jsonify({
        'status': 'healthy', 
        'service': 'PRP Calculator API',
        'version': '1.0.0'
    })
    
    # Add CORS headers
    response.headers['Access-Control-Allow-Origin'] = '*'
    response.headers['Access-Control-Allow-Methods'] = 'GET, OPTIONS'
    response.headers['Access-Control-Allow-Headers'] = 'Content-Type'
    
    return response